### ISSUE TITLE
NNX `YatNMN` spherical mode divides by norm without epsilon — NaN on zero-norm inputs/columns

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#19]
## Summary

In `src/nmn/nnx/layers/nmn.py:336-338`, the spherical normalization path divides inputs and kernel columns by their L2 norm without an additive `epsilon`. If any input row or kernel column has all-zero values (zero-padding tokens, dead neurons after pruning, initialization edge cases), the result is a NaN that propagates through the rest of the forward pass.

## Affected lines

```python
336:    if self.spherical:
337:       inputs = inputs / jnp.linalg.norm(inputs, axis=-1, keepdims=True)
338:       kernel = kernel / jnp.linalg.norm(kernel, axis=0, keepdims=True)
```

Compare with the Linen YatNMN, which is correctly guarded:

```python
# src/nmn/linen/nmn.py:136-138
if self.spherical:
    inputs = inputs / (jnp.linalg.norm(inputs, axis=-1, keepdims=True) + 1e-8)
    kernel = kernel / (jnp.linalg.norm(kernel, axis=-1, keepdims=True) + 1e-8)
```

The non-spherical `weight_normalized` path in the same NNX file (L334) also uses `+ 1e-8`, so the missing epsilon in the spherical branch is an oversight.

## Repro

```python
import jax.numpy as jnp
from flax import nnx
from nmn.nnx import YatNMN

layer = YatNMN(in_features=4, out_features=2, spherical=True, rngs=nnx.Rngs(0))
x = jnp.zeros((1, 4))   # zero-padding row
y = layer(x)
# -> NaN
```

## Suggested fix

Mirror Linen:

```python
if self.spherical:
    inputs = inputs / (jnp.linalg.norm(inputs, axis=-1, keepdims=True) + 1e-8)
    kernel = kernel / (jnp.linalg.norm(kernel, axis=0, keepdims=True) + 1e-8)
```

(For NNX the kernel shape is `(in_features, out_features)`, so `axis=0` correctly normalizes per output neuron — that part is fine.)
